HeterogeneousCore/CUDAUtilities/test/copyAsync_t.cpp

```diff
@@ -53,8 +55,9 @@ TEST_CASE("copyAsync", "[cudaMemTools]") {
 
       SECTION("Copy some") {
         cudautils::copyAsync(device, host_orig, 42, stream);
-        cuda::memory::async::copy(host.get(), device.get(), 42 * sizeof(int), stream);
         cudaCheck(cudaStreamSynchronize(stream));
+	cudaMemcpyAsync(host.get(), device.get(), 42 * sizeof(int), cudaMemcpyDeviceToHost, stream);
```

Comment:
These two lines seem incorrect:
  - `cudaMemcpyAsync` should come before `cudaStreamSynchronize`
  - `stream.synchronize();` seems spurious, possibly due to incorrect conflict resolution